internal/incident/incident.go

```diff
@@ -173,13 +182,30 @@ func (i *Incident) ProcessEvent(ctx context.Context, ev *event.Event) error {
 		}
 	case event.TypeAcknowledgementSet:
 		if err := i.processAcknowledgementEvent(ctx, tx, ev); err != nil {
+			if errors.Is(err, errSuperfluousAckEvent) {
+				// That ack error type indicates that the acknowledgement author was already a manager, thus
+				// we can safely ignore that event and return without even committing the DB transaction.
+				return nil
+			}
+
 			return err
 		}
 	}
 
-	notifications, err := i.addPendingNotifications(ctx, tx, ev, i.getRecipientsChannel(ev.Time))
-	if err != nil {
-		return err
+	var notifications []*NotificationEntry
+	// We only want to send notifications if one of the following conditions is met.
+	// 	- If the current incident is not yet muted or,
+	// 	- The current incident is already muted, but the source Object is unmuted with the ongoing event or,
+	// 	- The ongoing ev.Type isn't of type Downtime/Flapping-start and is neither ACK-set.
+	// Otherwise, we're going to generate suppressed/pending notifications history for all the remaining event types.
+	switch {
+	case !i.isMuted || !i.Object.IsMuted():
+		fallthrough
+	case ev.Type != event.TypeDowntimeStart && ev.Type != event.TypeAcknowledgementSet && ev.Type != event.TypeFlappingStart:
+		notifications, err = i.generateNotifications(ctx, tx, ev, i.getRecipientsChannel(ev.Time))
+		if err != nil {
+			return err
+		}
 	}
```

Comment:
The comment primarily repeats the code but doesn't say why it does what it does, especially why some event types are treated specially.

Also, I don't really understand the overall picture. This reads like this `switch` is supposed to determine whether notifications are sent, on the other hand, `generateNotifications()` determines internally whether notifications are suppressed?

Reply:
> This reads like this switch is supposed to determine whether notifications are sent, on the other hand, generateNotifications() determines internally whether notifications are suppressed?

No, this only determines whether `suppressed` or `pending` notifications should be generated at all, and `generateNotifications()` determines on its own whether the generated notifications should be of type `suppressed`.